src/cli/platform/sgx/crl.rs

```diff
@@ -22,10 +31,47 @@ pub struct CrlCache {}
 
 impl CrlCache {
     pub fn execute(self) -> anyhow::Result<ExitCode> {
-        let dir = sgx_cache_dir()?;
-        save_file(CERT_CRL, &dir)?;
-        save_file(PROCESSOR_CRL, &dir)?;
-        save_file(PLATFORM_CRL, &dir)?;
+        let mut dest_file = sgx_cache_dir()?;
+        dest_file.push("crls.der");
+
+        let dest_file = Path::new(&dest_file);
+        if dest_file.exists() {
+            std::fs::remove_file(dest_file)
+                .context(format!("deleting stale AMD CRLs file {dest_file:?}"))?;
+        }
+
+        let crls = vec![
+            fetch_file(CERT_CRL)
+                .context(format!("fetching {CERT_CRL}"))
+                .unwrap(),
+            fetch_file(PROCESSOR_CRL)
+                .context(format!("fetching {PROCESSOR_CRL}"))
+                .unwrap(),
+            fetch_file(PLATFORM_CRL)
+                .context(format!("fetching {PLATFORM_CRL}"))
+                .unwrap(),
+        ];
```

Comment:
Why create this Vec, when you don't need it? You could iterate over the file names and push to the Vec you need in the end. or you could use an array instead of a Vec.

Reply:
I tried, but the borrow checker complained about values not living long enough later on when trying to deserialize.